internal/worker_runner.go

```diff
@@ -157,6 +163,46 @@ func (wr *workerRunner) Run() {
 	}
 }
 
+func (wr *workerRunner) startLocationQuery(modem string) {
+	go func() {
+		wr.logger.Info().Msgf("Start query location data with every %.2f sec", wr.deviceSettings.LocationFrequencySecs)
+		for {
+			location, locationErr := wr.queryLocation(modem)
+			if locationErr == nil {
+
+				wr.signalsQueue.Enqueue(models.SignalData{
+					Timestamp: time.Now().UTC().UnixMilli(),
+					Name:      "longitude",
+					Value:     location.Longitude,
+				})
+
+				wr.signalsQueue.Enqueue(models.SignalData{
+					Timestamp: time.Now().UTC().UnixMilli(),
+					Name:      "latitude",
+					Value:     location.Latitude,
+				})
+
+				wr.signalsQueue.Enqueue(models.SignalData{
+					Timestamp: time.Now().UTC().UnixMilli(),
+					Name:      "hdop",
+					Value:     location.Hdop,
+				})
+
+				wr.signalsQueue.Enqueue(models.SignalData{
+					Timestamp: time.Now().UTC().UnixMilli(),
+					Name:      "nsat",
+					Value:     location.Nsat,
+				})
```

Comment:
I think location.Altitude also exists - consider adding it to this too

Reply:
Yep, tested, altitude exists